src/pocketbase/models/errors.py

```diff
@@ -1,13 +1,50 @@
 from typing import Any
 
+from httpx import Response
 
-class PocketbaseError(Exception):
+
+class PocketBaseError(Exception):
     def __init__(self, url: str, status: int, data: Any) -> None:
         self.url = url
         self.status = status
         self.data = data
 
     def __repr__(self) -> str:
-        return f"PocketbaseError(url={self.url},status={self.status},data={self.data})"
+        return f"{self.__class__.__name__}(url={self.url},status={self.status},data={self.data})"
 
     __str__ = __repr__
+
+    @classmethod
+    def raise_for_status(cls, response: Response) -> None:
+        if response.status_code == 400:
```

Comment:
https://pocketbase.io/docs/api-rules-and-filters/#:~:text=PocketBase%20API%20Rules%20act%20also%20as%20records%20filter!

Imho it's bad design by suabase, but `400` should also raise Unauthorized, even though it's not waht is meant. Or not? What do you think?

Reply:
Or maybe, the response json should be checked for if it was because of unsatisfied `createRule`, in which case raise Unauthorized, else do as is.

Reply:
This took a very long time, but now that I've sat down for it I've concluded that you cannot conclude that an unsatisfied `createRule` means `Unauthorized`. The `createRule` can indeed check for authentication but can also run validation on the new record itself. Thus the `400` makes sense and should not be converted to anything else in this case.